addons/mrp/models/mrp_production.py

```diff
@@ -1227,8 +1227,12 @@ def _set_qty_producing(self, pick_manual_consumption_moves=True):
 
             new_qty = float_round((self.qty_producing - self.qty_produced) * move.unit_factor, precision_rounding=move.product_uom.rounding)
             move._set_quantity_done(new_qty)
-            if not move.manual_consumption or pick_manual_consumption_moves:
-                move.picked = True
+            if self.env.context.get('mrp_display'):
+                if move.manual_consumption and pick_manual_consumption_moves:
+                    move.picked = False
+            else:
+                if not move.manual_consumption or pick_manual_consumption_moves:
+                    move.picked = True
```

Comment:
It can still be optimized, and I'm not sure about 'mrp_display' to check for shopfloor